jbpm-task-assigning/jbpm-task-assigning-core/src/test/java/org/jbpm/task/assigning/model/solver/TestSolver.java

```diff
@@ -0,0 +1,107 @@
+
+import java.util.List;
+
+import org.jbpm.task.assigning.BaseTaskAssigningTest;
+import org.jbpm.task.assigning.model.Task;
+import org.jbpm.task.assigning.model.TaskAssigningSolution;
+import org.jbpm.task.assigning.model.User;
+import org.junit.Test;
+import org.optaplanner.core.api.solver.Solver;
+
+import static org.jbpm.task.assigning.model.solver.TaskHelper.extractTaskList;
+import static org.jbpm.task.assigning.model.solver.TaskHelper.isPotentialOwner;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class TestSolver extends BaseTaskAssigningTest {
+
+    private static final long MILLISECONDS_TIME_SPENT_LIMIT = 20000;
+
+    @Test
+    public void startSolverAndSolution24Tasks8UsersTest() throws Exception {
+        testSolverStartAndSolution(MILLISECONDS_TIME_SPENT_LIMIT, _24TASKS_8USERS_SOLUTION);
+    }
+
+    @Test
+    public void startSolverAndSolution50Tasks5UsersTest() throws Exception {
+        testSolverStartAndSolution(MILLISECONDS_TIME_SPENT_LIMIT, _50TASKS_5USERS_SOLUTION);
+    }
+
+    @Test
+    public void startSolverAndSolution100Tasks5UsersTest() throws Exception {
+        testSolverStartAndSolution(MILLISECONDS_TIME_SPENT_LIMIT, _100TASKS_5USERS_SOLUTION);
+    }
+
+    @Test
+    public void startSolverAndSolution500Tasks5UsersTest() throws Exception {
+        testSolverStartAndSolution(MILLISECONDS_TIME_SPENT_LIMIT * 3, _500TASKS_20USERS_SOLUTION);
+    }
+
+    /**
+     * Tests that solver for the tasks assigning problem definition can be properly started, a solution can be produced,
+     * and that some minimal constrains are met by de solution.
+     */
+    private void testSolverStartAndSolution(long millisecondsSpentLimit, String solutionResource) throws Exception {
+        Solver<TaskAssigningSolution> solver = createNonDaemonSolver(millisecondsSpentLimit);
+        TaskAssigningSolution solution = readTaskAssigningSolution(solutionResource);
+        solution.getUserList().add(User.PLANNING_USER);
+        TaskAssigningSolution result = solver.solve(solution);
+        if (!result.getScore().isFeasible()) {
+            fail(String.format("With current problem definition and time spent of %s milliseconds it's expected " +
+                                       "that a feasible solution has been produced.", millisecondsSpentLimit));
+        }
+        assertConstraints(result);
+    }
+
+    /**
+     * Given a TaskAssigningSolution asserts the following constraints.
+     * <p>
+     * 1) All tasks are assigned to a user
+     * 2) The assigned user for a task is a potentialOwner for the task or the PLANNING_USER
+     * 3) All tasks are assigned.
+     * @param solution a solution.
+     */
+    private void assertConstraints(TaskAssigningSolution solution) {
+        int[] totalTasks = {0};
```

Comment:
If we always access just the 0 index of the array, do we need an array at all?

Reply:
this variable is used to collect information inside a solution.getUserList().forEach at the time we do other processing.  So I think it's ok.